java/Cherepnin_Oleh/src/com/zoolatech/lecture4/tasks/_2/Tree.java

```diff
@@ -0,0 +1,88 @@
+package com.zoolatech.lecture4.tasks._2;
+
+public class Tree {
+    private Node head;
+
+    public Tree() {
+    }
+
+    public boolean add(int value) {
+        Node newNode = new Node(value);
+
+        if (head == null) {
+            head = newNode;
+            return true;
+        }
+
+        Node current = this.head;
+        do {
+            if (value == current.getValue()) {
```

Comment:
just a note: binary trees usually allow to have duplicate values, but let's keep it as is